src/lib/comms/sol-coap.c

```diff
@@ -1614,3 +1622,20 @@ sol_coap_unobserve_server(struct sol_coap_server *server, const struct sol_netwo
 
     return -ENOENT;
 }
+
+SOL_API void
+sol_coap_packet_debug(struct sol_coap_packet *pkt)
```

Comment:
you can also use the SOL_LOG debug stuff, it can be disabled and we could remove it at this location and also in the header file (empty `static inline`)